### PR TITLE
Update version and recipe (include PL-Lightning).

### PR DESCRIPTION
**Pull request by vincentmr**

### CODE REVIEW

**Comment by BastianZim** [comment]
Hi @vincentmr is this ready for review or are you just moving it over?

**Comment by vincentmr** [comment]
> Hi @vincentmr is this ready for review or are you just moving it over?

Yes, ready for review. I pretty much just transferred as is rerendered.

**Comment by BastianZim** [comment]
Ok, thank you. Can you please still answer the questions in https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16 as I need to update the infra?

**Comment by vincentmr** [comment]
> Can you please still answer the questions in https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16 as I need to update the infra?

Hi @BastianZim , sorry I thought I did. Would you mind pointing it out if I missed any?

**Comment by BastianZim** [comment]
No worries, just the open ones:

https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16#discussion_r1122307694
https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16#discussion_r1122310359
https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16#discussion_r1122315689

**Comment by vincentmr** [comment]
> No worries, just the open ones:

My answers where "Pending", basically since you posted the questions. I had not realised you could not see them. My bad.

**Comment by BastianZim** [comment]
Hi @ocefpaf sorry for the ping because this is a slightly larger question. Do you have any experience with multi-output recipes? This is a slightly larger change so I was wondering if I could get a second pair of eyes for a review if this is correct?
The packages have circular dependencies so we are building both in this feedstock now instead of two. Previously, this package was a single output so I just want to make sure that we're doing this correctly. 
The other feedstock is in: https://github.com/conda-forge/pennylane-lightning-feedstock/pull/16

**Comment by ocefpaf** [comment]
Looks OK but, to be honest, the best way to check is to build it locally and inspect the artifacts. Multiple outputs still bites me every now and then too.

**Comment by BastianZim** [comment]
Thank you! I believe we did that and it works (@vincentmr can you confirm?) I'm just not sure about the empty top-level because that normally needs to be a meta package but in this case it works.

**Comment by vincentmr** [comment]
> I believe we did that and it works (@vincentmr can you confirm?)
Yes, I built locally and in the Lightning feedstock I had turned on storing artifacts on Azure, which I could then download and install, and that was also working.

**Comment by BastianZim** [comment]
OK sounds good. Will make sure that the outputs work and then we can merge.

**Comment by BastianZim** [comment]
Ahh wait you renamed the packages so nothing required.

**Comment by leofang** [comment]
It'd be nice if you could archive pennylane-lightning-feedstock 🙂 I'm evaluating how to enable Lightning's GPU support, and it took me a while to understand that feedstock is abandoned...

**Comment by BastianZim** [comment]
Yeah, I’ll put up an issue. Not sure if archiving is the best because we will probably use it later down the road.

**Comment by leofang** [comment]
Thanks!

**Comment by vincentmr** [comment]
> It'd be nice if you could archive pennylane-lightning-feedstock 🙂 I'm evaluating how to enable Lightning's GPU support, and it took me a while to understand that feedstock is abandoned...

I have a PR on this [here](https://github.com/conda-forge/staged-recipes/pull/22296).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,57 +1,127 @@
...
+        - libgomp      # [linux]
+        - llvm-openmp  # [osx]
+        - pip
+        - pybind11
+        - python
+        - setuptools <=64.0.2 # version requirement from cvxpy
+      run:
+        - ninja
+        - numpy
+        - python
```
Comment (leofang) [inline recipe/meta.yaml:83]: @conda-forge/pennylane I'm trying to debug this issue: https://github.com/conda-forge/pennylane-lightning-gpu-feedstock/issues/1#issuecomment-1494761395, and I guess this is the reason:
```suggestion
      run:
        - ninja
        - numpy
        - python
      run_constrained:
        - pennylane-lightning <=0.24.0
```
Without introducing a conflict between the old/new formats, the conda solver would take both formats into account, and depending on the solver internal (conda vs mamba) either format could be picked.

Comment (leofang) [reply recipe/meta.yaml:83]: (fixed the upper bound lol)

Comment (vincentmr) [reply recipe/meta.yaml:83]: I'm not familiar with this command. So this says that `pennylane-lightning-core` may depend on `pennylane-lightning`, but only versions `<0.0a0`, which are none of them. So it's not possible to depend on both.

Comment (leofang) [reply recipe/meta.yaml:83]: Exactly 🙂 This is a common trick to ensure that two packages cannot coexist in the same environment (it's a solver conflict -> error). We used it too: https://github.com/conda-forge/cuquantum-feedstock/blob/0685dca47744279676ad40fb518e575d47c8aade/recipe/meta.yaml#L168-L172. There're other ways to do this, but this is the simplest.

Comment (vincentmr) [reply recipe/meta.yaml:83]: I'll make a PR for this, unless you were just about to open one?

Comment (leofang) [reply recipe/meta.yaml:83]: Would you mind doing it for me, Vincent? I don't have maintainer access here, so I can't just ask the bot to open a draft PR for me to edit 😅

**Pull request merged**